pkg/test/environment.go

```diff
@@ -873,7 +873,7 @@ func (t TestNodeConfig) MetricReceivers() []string {
 	res := []string{}
 	if t.Metrics.Enabled {
 		res = append(res, "prometheus/self")
-		if t.Metrics.Spec.HostMetrics {
+		if t.Metrics.Spec.HostMetrics != nil && *t.Metrics.Spec.HostMetrics {
```

Comment:
Minor nit but it's probably cleaner to use something like
```go
if lo.FromPtrOr(t.Metrics.Spec.HostMetrics, false) {
```

Reply:
agreed, keep the nits coming